### PR TITLE
Insert test section into CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by eugene70**
Inserted a section about test into CONTRIBUTING.md as following:

  "We support both JUnit 4 and JUnit 5 for testing, but we recommend to use JUnit 5."

Also changed the protocol of all links to 'https' instead of 'http'.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -71,6 +71,24 @@ After importing the project, import the IDE settings as well.
 You can configure your build tool and IDE to add `-parameters` javac option.
 Please refer to [Configure `-parameters` javac option](http://line.github.io/armeria/setup.html#configure-parameters-javac-option) for more information.
 
+### Use JUnit5 instead of JUnit4 for test
+
+We support both JUnit4 and JUnit5 for test. but, we recommend to use JUnit5.
```
Comment (ikhoon) [inline CONTRIBUTING.md]: Could you add links for them? 😁
* https://junit.org/junit4/
* https://junit.org/junit5/

Comment (eugene70) [reply CONTRIBUTING.md]: Sure! 
I did it.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -69,7 +69,25 @@ After importing the project, import the IDE settings as well.
 ### Configure `-parameters` javac option 
 
 You can configure your build tool and IDE to add `-parameters` javac option.
-Please refer to [Configure `-parameters` javac option](http://line.github.io/armeria/setup.html#configure-parameters-javac-option) for more information.
+Please refer to [Configure `-parameters` javac option](https://line.github.io/armeria/setup.html#configure-parameters-javac-option) for more information.
+
+### Use JUnit5 instead of JUnit4 for test
```
Comment (trustin) [inline CONTRIBUTING.md]: - JUnit5 -> JUnit 5
- JUnit4 -> JUnit 4
- for test -> for testing

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -69,7 +69,25 @@ After importing the project, import the IDE settings as well.
...
+
+We support both [JUnit4](https://junit.org/junit4/) and [JUnit5](https://junit.org/junit5/) for test. but, we recommend to use JUnit5.
```
Comment (trustin) [inline CONTRIBUTING.md]: - `test. but, we ` -> `testing, but we `
- JUnit5 -> JUnit 5
- JUnit4 -> JUnit 4

Comment (eugene70) [reply CONTRIBUTING.md]: I fixed it, thank you.

**Pull request merged**